### PR TITLE
Add a thin wrapper component for default DateRange facet.

### PR DESCRIPTION
**Pull request by olamothe**
TimespanFacet simply configures a standard RangeFacet with prebuilt date range to filter documents on the last day, week, month and year.

I also added this component in most of the default search pages, and removed the standard facet on month and year.

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

**Comment by olamothe** [comment]
@amoreauCoveo for the documentation 👀

File: src/ui/TimespanFacet/TimespanFacet.ts [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+  constructor(public element: HTMLElement, public options?: ITimespanFacetOptions, bindings?: IComponentBindings) {
+    super(element, TimespanFacet.ID, bindings);
+
+    this.options = ComponentOptions.initComponentOptions(element, TimespanFacet, options);
+    this.buildFacet();
+  }
+
+  /**
+   * Allow to set new ranges programmatically.
```
Comment (amoreauCoveo) [inline src/ui/TimespanFacet/TimespanFacet.ts]: Sets a new range for the component.

File: src/ui/TimespanFacet/TimespanFacet.ts [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+   *
+   * Destroy the old {@link TimespanFacet.facet} if needed, and recreate the component with the new ranges.
```
Comment (amoreauCoveo) [inline src/ui/TimespanFacet/TimespanFacet.ts]: You might need to delete the old {@link TimespanFacet.facet} and recreate the component the new range.

File: src/ui/TimespanFacet/TimespanFacet.ts [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+   */
+  public set ranges(ranges: IRangeValue[]) {
+    this.rangeValues = ranges;
+    this.buildFacet();
+  }
+
+  /**
+   * The current date ranges that the facet uses to query the index.
```
Comment (amoreauCoveo) [inline src/ui/TimespanFacet/TimespanFacet.ts]: Returns the current range the facet uses to query the index.

File: src/ui/TimespanFacet/TimespanFacet.ts [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+   */
+  public get ranges() {
+    return this.rangeValues;
+  }
+
+  /**
+   * The underlying {@link FacetRange} component
```
Comment (amoreauCoveo) [inline src/ui/TimespanFacet/TimespanFacet.ts]: Returns the underlying {@link FacetRange} component associated to the Timespan Facet.

File: src/ui/Base/InitializationPlaceholder.ts [hunk 5]
```diff
@@ -314,6 +314,7 @@ export class InitializationPlaceholder {
   private getAllFacetsElements(): HTMLElement[] {
     let facetElements = $$(this.root).findAll('.CoveoFacet');
```
Comment (dms1lva) [inline src/ui/Base/InitializationPlaceholder.ts]: Here we could use an array for all the selectors we want to catch and then build a selector from this array. Would be more concise / easier to add more selectors.

**Pull request merged**